### DISCUSSION TITLE
Dear Diary.... JSON is not listening..

### DISCUSSION THREAD

**Discussion by JSchugt** [nss-day-cohort-47/C47-Discussions#36] [Help Me] [answered]
I am attempting to pull journal data from the local JSON server and send it into an array to be sorted.  The raw data is being pulled from the server but when I attempt to push it on to  journal[]  I get an error with the sort function " journal.sort is not a function".   I've attempted to iterate over the JSON data and push it, I've attempted saving it directly to journal at instantiation.  But always have the same errors.  

When journal is written in the same file as an array it works and displays the data.

**Comment by ardisrtilson** [comment]
Hey Jeremiah, I'd like to see your code. You can either come to my breakout room or post it here!

**Reply by JSchugt** [reply]
#redIsAwesome

**Reply by BryanNilsen** [reply]
@JSchugt - can you post a link to the code and a resolution if you've solved it so we can see how this worked out??

**Reply by ardisrtilson** [reply]
I'd like to see Jeremiah's explanation, but I'll just say the key to some of the problems I've been seeing is to mind the data types. Look at the data you pull from the server and ask yourself, "Is this a string? Is this an array? Is this an object? Is this an object full of arrays? Is this an array full of objects? Etc. Then remember that we have array methods, which are for arrays, and string methods, which are for strings. If we've got an object, we can't use array methods, for instance, and so on.

**Comment by JSchugt** [comment, marked as answer]
-- Edit move response to comment for resolution. 
So with the help of @ardisrtilson, The first issue was trying to pull the data into the journal from the JSON server. Initially the set up has an array const journal = [ { id: 1, date: "07/24/2025", concept: "HTML & CSS", entry: "We talked about HTML components and how to make grid layouts with Flexbox in CSS.", mood: "Ok" } ]

Which using this in the JournalDataProvider.js worked and displayed the entry on the DOM.

My first attempt after pulling the data from the JSON server
```
import {getJournalEntry} from "./data/EntryManager.js";

import { JournalEntryComponent } from "./JournalEntry.js";

const journal = getJournalEntry();

console.log(getJournalEntry())

for(const entry of getJournalEntry().then(response => {return response.json()})){

 journal.push(entry);
}
```
getJournalEntry() returns a JSON object that contains the data from the journal entries, but I kept receiving the error when I attempted to sort journal "journal.sort(....) is not a function or is not itterable". I tried looping over the getEntryData to push that onto the journal array and still continued to get errors.

After Red's assistance, the first problem was that it was not in array format. So going to go the source of the fetch, we created a global variable to hold the data from the fetch in an array
let arrayResponse = [];

then call a function userJournalEntries()

```
export const useJournalEntries = () => {

getJournalEntry().then(() => {

    console.log(arrayResponse)

    const sortedByDate = arrayResponse.sort(

        (currentEntry, nextEntry) =>

            Date.parse(currentEntry.date) - Date.parse(nextEntry.date)

    )

    return sortedByDate

})

return arrayResponse;
}
```

with the function getJournalEntry()

```
export const getJournalEntry = () => {

return fetch("http://localhost:8088/journalEntry")

    .then(response => response.json())

    .then(parsedResponse => {

        // do something with response here

        arrayResponse = parsedResponse;

    })
}
```

These together pulled the data from the server, sorted the data and returned an array with the entries in order.
Below is the link to the github repo where the code is being hosted.
